src/data_router.py

```diff
@@ -0,0 +1,71 @@
+import glob
+import logging
+import multiprocessing
+import os
+import random
+from flask import json
+
+from config import RasaNLUConfig
+from rasa_nlu.train import do_train
+
+
+class DataRouter(object):
+    def __init__(self, config, interpreter, emulator):
+        self.ID = random.random()
+        self.config = config
+        self.interpreter = interpreter
+        self.emulator = emulator
+        self.logfile = config['write'].replace(".json", "-{}.json".format(os.getpid()))
+        self.responses = self.create_query_logger(self.logfile)
+        self.train_proc = None
+        self.model_dir = config['path']
+        self.token = config['token']
+
+    def create_query_logger(self, path):
+        logger = logging.getLogger('query-logger')
+        logger.setLevel(logging.INFO)
+        ch = logging.FileHandler(path)
+        ch.setFormatter(logging.Formatter('%(message)s'))
+        logger.propagate = False
+        logger.addHandler(ch)
+        return logger
+
+    def extract(self, data):
+        return self.emulator.normalise_request_json(data)
+
+    def parse(self, text):
+        result = self.interpreter.parse(text)
+        self.responses.info(json.dumps(result, sort_keys=True))
+        return result
+
+    def format(self, data):
+        return self.emulator.normalise_response_json(data)
+
+    def get_status(self):
+        if self.train_proc is not None:
+            training = self.train_proc.is_alive()
+        else:
+            training = False
+
+        models = glob.glob(os.path.join(self.model_dir, 'model*'))
+        return {
+            "training": training,
+            "available_models": models
+        }
+
+    def start_train_proc(self, data):
+        logging.info("starting train")
+        if self.train_proc is not None and self.train_proc.is_alive():
```

Comment:
Yes you are right. I think that is the only simple solution since otherwise we would need to communicate between the processes. The only thing I am worried about is, that this endpoint is secured in the same way as the others meaning that potentially everyone who has the auth token can train models there.